src/main/java/emissary/util/DateTimeFormatParser.java

```diff
@@ -0,0 +1,211 @@
+package emissary.util;
+
+import emissary.config.Configurator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ */
+public class DateTimeFormatParser {
+
+    protected static final List<DateTimeFormatter> dateTimeZoneFormats = new ArrayList<>();
+    protected static final List<DateTimeFormatter> dateTimeOffsetFormats = new ArrayList<>();
+    protected static final List<DateTimeFormatter> dateTimeFormats = new ArrayList<>();
+    protected static final List<DateTimeFormatter> dateFormats = new ArrayList<>();
+
+    private static final ZoneId GMT = ZoneId.of("GMT");
+
+    protected static final Logger logger = LoggerFactory.getLogger(DateTimeFormatParserLegacy.class);
+
+    static {
+        configure();
+    }
+
+    protected static void configure() {
+
+        Configurator configG;
+        try {
+            configG = emissary.config.ConfigUtil.getConfigInfo(DateTimeFormatParser.class);
+        } catch (IOException e) {
+            logger.error("Cannot open default config file", e);
+            return;
+        }
+
+        for (final String dfentry : configG.findEntries("DATE_TIME_ZONE_FORMAT")) {
```

Comment:
I made this change to lessen the amount of code, seems to work. Could maybe even be further lessened with a Map<String, List>?

```suggestion
protected static void configure() {

        Configurator configG;
        try {
            configG = emissary.config.ConfigUtil.getConfigInfo(DateTimeFormatParser.class);
        } catch (IOException e) {
            logger.error("Cannot open default config file", e);
            return;
        }

        loadDateTimeEntries(configG, "DATE_TIME_ZONE_FORMAT", dateTimeZoneFormats);
        loadDateTimeEntries(configG, "DATE_TIME_OFFSET_FORMAT", dateTimeOffsetFormats);
        loadDateTimeEntries(configG, "DATE_TIME_FORMAT", dateTimeFormats);
        loadDateTimeEntries(configG, "DATE_FORMAT", dateFormats);

    }

    private static void loadDateTimeEntries(Configurator configG, String entryType, List<DateTimeFormatter> dateFormats) {
        for (final String dfentry : configG.findEntries(entryType)) {
            try {
                DateTimeFormatter initialSdf =
                        new DateTimeFormatterBuilder().parseCaseInsensitive().appendPattern(dfentry).toFormatter();
                if (entryType.equals("DATE_TIME_ZONE_FORMAT")) {
                    initialSdf = initialSdf.withZone(GMT);
                }
                final DateTimeFormatter sdf = initialSdf;
                dateFormats.add(sdf);
            } catch (Exception ex) {
                logger.debug("{} entry '{}' cannot be parsed", entryType, dfentry, ex);
            }
        }
        logger.debug("Loaded {} {} entries", dateTimeZoneFormats.size(), entryType);
    }        
```